tools/cli/commands/draft.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * The `jetpack draft enable|disable|new` command
+ */
+
+/**
+ * External dependencies
+ */
+import path from 'path';
+import inquirer from 'inquirer';
+import chalk from 'chalk';
+import fs from 'fs';
+import child_process from 'child_process';
+
+/**
+ * Internal dependencies
+ */
+import { chalkJetpackGreen } from '../helpers/styling';
+
+/**
+ * Returns the path to the .jetpack-draft file
+ *
+ * @returns {string} - the draft file full path
+ */
+function getDraftFile() {
+	return path.join( process.cwd(), '.jetpack-draft' );
+}
+
+/**
+ * Enable draft mode.
+ *
+ * @param {object} argv - The argv for the command line.
+ */
+export async function draftEnable( argv ) {
+	try {
+		if ( fs.existsSync( getDraftFile() ) ) {
+			console.log( chalk.yellow( 'Draft mode is already enabled.' ) );
+			return;
+		}
+
+		fs.closeSync( fs.openSync( getDraftFile(), 'w' ) );
+
+		console.log(
+			chalkJetpackGreen( 'You are now in draft mode. No nags for you, but be careful.' )
+		);
+
+		if ( argv.v ) {
+			console.log( argv );
+		}
+	} catch ( e ) {
+		console.error( chalk.red( 'Uh oh! ' + e.message ) );
+		console.log( argv );
+		process.exit( 1 );
+	}
+}
+
+/**
+ * Enable draft mode.
+ *
+ * @param {object} argv - The argv for the command line.
+ */
+export async function draftDisable( argv ) {
+	try {
+		if ( ! fs.existsSync( getDraftFile() ) ) {
+			console.log( chalk.yellow( 'Draft mode is not enabled.' ) );
+			return;
+		}
+
+		fs.unlinkSync( getDraftFile() );
+
+		console.log(
+			chalkJetpackGreen(
+				'You have exited draft mode. Jetpack will now run standard pre-commit and pre-push hooks. '
+			)
+		);
+
+		const preCommitAnswers = await inquirer.prompt( [
+			{
+				type: 'confirm',
+				name: 'runPreCommit',
+				default: false,
+				message: 'Would you like to run pre-commit checks now?',
```

Comment:
How much point is there to this, since most of the checks look at the difference between the git index and the commit? Chances are by the time you get here you've already made commits.